### PR TITLE
Compressed source test and cleanups

### PR DESCRIPTION
**Pull request by dhalperi**
R: @jkff @bjchambers

### CODE REVIEW

File: sdks/java/core/src/test/java/org/apache/beam/sdk/io/TextIOTest.java [hunk 1]
```diff
@@ -638,6 +638,33 @@ public void testGZIPReadWhenUncompressed() throws Exception {
     p.run();
   }
 
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testGZIPReadWhenCompressed() throws Exception {
+    final int numLines = 1000;
```
Comment (kennknowles) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/TextIOTest.java]: Same test useful for BZIP and AUTO.

Comment (kennknowles) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/TextIOTest.java]: A comment about why 1000 so no one tries to reduce it.

File: sdks/java/core/src/test/java/org/apache/beam/sdk/io/TextIOTest.java [hunk 2]
```diff
@@ -638,6 +638,33 @@ public void testGZIPReadWhenUncompressed() throws Exception {
...
+    List<String> expected = new ArrayList<>();
+    try (PrintStream writer = new PrintStream(
+        new GZIPOutputStream(new FileOutputStream(tmpFile)))) {
+      for (int i = 0; i < numLines; ++i) {
+        String line = "word" + i;
+        writer.println(line);
+        expected.add(line);
+      }
+    }
+
```
Comment (kennknowles) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/io/TextIOTest.java]: Assert that the compressed size is actually less than uncompressed size.

**Comment by kennknowles** [comment]
LGTM with some testing comments. The only one I'd block merge on is the assertion that the compressed size is smaller than uncompressed size.

**Comment by dhalperi** [comment]
Merging based on Kenn's review and similar on https://github.com/GoogleCloudPlatform/DataflowJavaSDK/issues/357

**Pull request closed without merging**